quickwit-config/src/config.rs

```diff
@@ -332,9 +395,18 @@ impl QuickwitConfig {
             });
         }
         if let Some(advertise_addr) = &self.advertise_address {
-            return Ok(advertise_addr.clone());
+            return advertise_addr.parse().map(|addr| {
+                info!(advertise_address=%advertise_addr, "Using advertise address from config file.");
+                addr
+            }).with_context(|| {
+                format!(
+                    "Failed to parse advertise address `{advertise_addr}` read from \
+                     config file."
+                )
+            });
         }
-        if self.listen_address.is_unspecified() {
+        // This is safe, check the declaration.
```

Comment:
What do you mean?

Reply:
Artifact comment from my previous attempt to do something stupid. Not relevant anymore. Removed